main.cpp

```diff
@@ -92,6 +92,129 @@ good to go!
  */
 
 #include <iostream>
+
+ /*4) make them return the correct primitive type. e.g. if you're implementing the FloatType::add function, your implementation would start like this:
+        float FloatType::add( float lhs, float rhs ) */
+
+#include <math.h>
+#ifdef INFINITY
+#endif
+
+
+///////// FLOAT  ...............
+///////// 
+class FloatType 
+{
+    public:
+        float lhs, rhs;
+        float add(float, float);
+        float subtract(float, float);
+        float multiply(float, float);
+        float divide(float, float);
+};
+
+float FloatType::add(float a, float b)
+{
+    return a + b;
+}
+
+float FloatType::subtract(float a, float b)
+{
+    return a - b;
+}
+
+float FloatType::multiply(float a, float b)
+{
+    return a * b;
+}
+
+float FloatType::divide(float a, float b)
+{
+    if (b<1e-7f && b>-1e-7f)
+        {
+        std::cout <<std::endl<<"warning, floating point division by zero returns 'inf' !" <<std::endl;
+            return(a/b);
+        }
+    return a/b;
+}
```

Comment:
1) `if( b == 0.f )` is the correct solution here
2) add space around any math operators and any `<<` to make your code more readable.
3) the indentation needs fixing
- [ ] revise